sandbox/cgroup/scanning.go

```diff
@@ -28,11 +28,18 @@ import (
 	"github.com/snapcore/snapd/snap/naming"
 )
 
+const (
+	uuidPattern = `[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}`
+)
+
 var (
+
 	// string that looks like a hook security tag
-	roughHookTagPattern = regexp.MustCompile(`snap\.[^.]+\.hook\.[^.]+`)
+	roughHookTagPattern         = regexp.MustCompile(`snap\.[^.]+\.hook\.[^.]+`)
+	roughHookTagPatternWithUUID = regexp.MustCompile(`(snap\.[^.]+\.hook\.[^.]+)(-` + uuidPattern + `)`)
 	// string that looks like an app security tag
-	roughAppTagPattern = regexp.MustCompile(`snap\.[^.]+\.[^.]+`)
+	roughAppTagPattern         = regexp.MustCompile(`snap\.[^.]+\.[^.]+`)
+	roughAppTagPatternWithUUID = regexp.MustCompile(`(snap\.[^.]+\.[^.]+)(-` + uuidPattern + `)`)
```

Comment:
Is there any particular reason why two match groups are used in these instances? As opposed to something like
```
- `(snap\.[^.]+\.hook\.[^.]+)(-` + uuidPattern + `)`
+ `snap\.[^.]+\.hook\.[^.]+-` + uuidPattern
...
- `(snap\.[^.]+\.[^.]+)(-` + uuidPattern + `)`
+ `snap\.[^.]+\.[^.]+-` + uuidPattern
```
What you have clearly works, just curious :)

Reply:
It's been a while since I wrote this commit, but probably because I thought it was a clear way to extract the tag from the matching string. But I guess since the UUID portion has to be a fixed length, we could do this with `match[:len(match) - 37 /* Length of UUID + '-' */]`.